org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/DebugAdapter.java

```diff
@@ -376,8 +395,10 @@ public DebugAdapter(BiConsumer<Events.DebugEvent, Boolean> consumer, IProviderCo
         if (thread != null) {
             allThreadsContinued = false;
             thread.resume();
+            this.variableHandler.recyclableThreads(thread);
         } else {
             this.debugSession.resume();
+            this.variableHandler.recyclableAllObject();
```

Comment:
I'm afraid the else branch will never be hit because there will always be a thread specified by VSCode. We probably need to think of other ways of recycling like checking whether all threads are running.